lib/index.js

```diff
@@ -469,13 +500,33 @@ NodeHttpXHR.prototype.send = function (data) {
     this._resp = resp;
     this._responseText = '';
 
-    resp.setEncoding('utf8');
+    // var contentType = resp.headers['content-type'];
+    // TODO: adjust responseType from content-type header if applicable
+
+    // from Node API docs: The encoding argument is optional and only applies when chunk is a string. Defaults to 'utf8'.
+    if (this._responseType === 'text' || this._responseType === '') {
+      resp.setEncoding('utf8');
+    }
+
     resp.on('data', function onData(chunk) {
-      this._responseText += chunk;
+
+      if (typeof chunk === 'string') {
```

Comment:
`switch (typeof chunk)` ?

Reply:
in this case i don't think it makes sense to use a switch-case.

there isn't a generic amount of possible values. this is the logic you need, either it's string or it's buffer. there aren't any other possibilities. using a switch-case here just makes it harder to read or understand imo, and implies there might be more possibiltities to implement, which isn't the case.

Reply:
Fair enough 👍